rotkehlchen/tests/unit/decoders/test_thegraph.py

```diff
@@ -0,0 +1,174 @@
+    events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=ethereum_inquirer,
+        database=database,
+        tx_hash=tx_hash,
+    )
+    timestamp = TimestampMS(1691771855000)
+    expected_events = [
+        EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=0,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(amount=FVal('0.00307607001551556')),
+            location_label=ADDY_USER,
+            notes='Burned 0.00307607001551556 ETH for gas',
+            counterparty=CPT_GAS,
+        ), EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=297,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.NONE,
+            asset=A_GRT,
+            balance=Balance(),
+            location_label=ADDY_USER,
+            notes=(
+                'Undelegate 1003.70342593701668535 GRT from'
+                ' indexer 0x6125eA331851367716beE301ECDe7F38A7E429e7. Lock expires in 983 seconds'
+            ),
+            counterparty=CPT_THEGRAPH,
+            address=ADDY_THEGRAPH,
+        ),
+    ]
+    assert expected_events == events
+
+
+@pytest.mark.vcr()
+@pytest.mark.parametrize('ethereum_accounts', [[ADDY_USER]])
+def test_thegraph_delegated_withdrawn(database, ethereum_inquirer):
+    tx_hash = deserialize_evm_tx_hash('0x49307751de5ba4cf98fccbdd1ab8387fd60a7ce120800212c216bf0a6a04acfa')  # noqa: E501
+    events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=ethereum_inquirer,
+        database=database,
+        tx_hash=tx_hash,
+    )
+    timestamp = TimestampMS(1694577371000)
+    expected_events = [
+        EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=0,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(amount=FVal('0.000651667321615926')),
+            location_label=ADDY_USER,
+            notes='Burned 0.000651667321615926 ETH for gas',
+            counterparty=CPT_GAS,
+        ),
+        EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=208,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.STAKING,
+            event_subtype=HistoryEventSubType.REMOVE_ASSET,
+            asset=A_GRT,
+            balance=Balance(amount=FVal('1003.70342593701668535')),
+            location_label=ADDY_USER,
+            notes=(
+                'Withdraw 1003.70342593701668535 GRT'
+                ' from indexer 0x6125eA331851367716beE301ECDe7F38A7E429e7'
```

Comment:
This is also a nitpick and can't enforce from linter but I have notices when you split a sentence string across multiple lines you end the line without a space and start the new with a space. Can you try to be doing the opposite? It's how we do it in the rest of the code.

Really minor nitpick to keep in mind. Here and in all other places you do it.

```suggestion
                'Withdraw 1003.70342593701668535 GRT '
                'from indexer 0x6125eA331851367716beE301ECDe7F38A7E429e7'
```